pkg/cloudprovider/providers/aws/aws_loadbalancer.go

```diff
@@ -308,6 +308,49 @@ func (c *Cloud) ensureLoadBalancerv2(namespacedName types.NamespacedName, loadBa
 			}
 		}
 
+		// Whether the ELB was new or existing, sync attributes regardless. This accounts for things
+		// that cannot be specified at the time of creation and can only be modified after the fact,
+		// e.g. idle connection timeout.
+		if len(loadBalancerAttributes) > 0 {
+			describeAttributesRequest := &elbv2.DescribeLoadBalancerAttributesInput{
+				LoadBalancerArn: loadBalancer.LoadBalancerArn,
+			}
+			describeAttributesOutput, err := c.elbv2.DescribeLoadBalancerAttributes(describeAttributesRequest)
+			if err != nil {
+				glog.Warning("Unable to retrieve load balancer attributes during attribute sync")
+				return nil, err
+			}
+
+			foundAttributes := &describeAttributesOutput.Attributes
+
+			changedAttributes := []*elbv2.LoadBalancerAttribute{}
+
+			// Identify to be changed attributes
+			for _, foundAttribute := range *foundAttributes {
+				if targetValue, ok := loadBalancerAttributes[*foundAttribute.Key]; ok {
```

Comment:
You should add an `else` case for `if !ok`, meaning the annotation for the attribute was deleted.

From [the docs](https://docs.aws.amazon.com/elasticloadbalancing/latest/APIReference/API_ModifyLoadBalancerAttributes.html):
> Any existing attributes that you do not modify retain their current values.

If the annotation gets removed, we would want to set the value to `"false"` and add it to `changedAttributes`.

Reply:
Wouldn't that mean that we set any value to `false`, also the attributes k8s doesn't know about and which might have a different default (e.g. `routing.http2.enabled`, while only being a valid attribute for ALBs defaults to `true`).

Instead I set now `load_balancing.cross_zone.enabled` to `false` by default, so a set annotation will change it, but an eventually removed annotation changes it back to `false`.